probe_src/libprobe/Makefile

```diff
@@ -1,7 +1,7 @@
 SOURCE_VERSION ?= $(shell git rev-parse --short HEAD)
-DBGCFLAGS ?= -DSOURCE_VERSION=\"$(SOURCE_VERSION)\" -Wno-cast-function-type -Wno-array-parameter -ldl -Wall -Wextra -pthread -Og -g
-OPTCFLAGS ?= -DSOURCE_VERSION=\"$(SOURCE_VERSION)\" -Wno-cast-function-type -Wno-array-parameter -ldl -Wall -Wextra -pthread -O3 -DNDEBUG
-LIBCFLAGS ?= -fPIC -shared
+DBGCFLAGS ?= -DSOURCE_VERSION=\"$(SOURCE_VERSION)\" -Wno-cast-function-type -Wno-array-parameter -ffreestanding -Wl,--as-needed -ldl -Wall -Wextra -pthread -Og -g
+OPTCFLAGS ?= -DSOURCE_VERSION=\"$(SOURCE_VERSION)\" -Wno-cast-function-type -Wno-array-parameter -ffreestanding -Wl,--as-needed -Wall -Wextra -pthread -O3 -DNDEBUG
```

Comment:
```suggestion
CFLAGS ?= -DSOURCE_VERSION=\"$(SOURCE_VERSION)\" -Wno-cast-function-type -Wno-array-parameter -ffreestanding -Wl,--as-needed -Wall -Wextra -pthread
DBGCFLAGS ?= $(CFLAGS) -Og -g
OPTCFLAGS ?= $(CFLAGS) -O3 -DNDEBUG
```

One has `-ldl` and the other doesn't? I assume this is a typo. While here, we should refactor this so its easier to get right.

Reply:
Yes, that was a typo, good catch.